src/Features/Core/Portable/UseCoalesceExpression/AbstractUseCoalesceExpressionDiagnosticAnalyzer.cs

```diff
@@ -24,6 +25,9 @@ protected AbstractUseCoalesceExpressionDiagnosticAnalyzer()
         {
         }
 
+        public override bool OpenFileOnly(Workspace workspace) => false;
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
```

Comment:
IIRC, we could not use SemanticSpan here.  That's because the analyzer engine will not update 'AdditionalLocations' on Diagnostics properly if you use that.